fdbserver/DataDistribution.actor.cpp

```diff
@@ -3615,12 +3617,19 @@ ACTOR Future<Void> checkAndRemoveInvalidLocalityAddr(DDTeamCollection* self) {
 				if (!existingAddrs.count(*addr)) {
 					// The address no longer has a worker
 					addr = self->invalidLocalityAddr.erase(addr);
+					hasCorrectedLocality = true;
```

Comment:
Why does it need to rerun the recruiting if the incorrect address was removed? I mean in that case there is no change of the topology logically, because the removed process was not used anyways.

Reply:
invalidLocalityAddr is more like the excluded address.
When its locality is correct, it should no longer be excluded.